deployment/promotion/promote.py

```diff
@@ -9,396 +9,157 @@
-        'uses_flask': cr['uses_flask']
-    }
-    return post(auth, path, build)
-
-
-def update_build_message(auth, source_build, dest_build):
-    message = source_build['message']
-    print("Setting %d byte build message" % len(message))
-    put(auth, "/api/v1/deployment/builds/{}/message".format(dest_build['id']), message)
-
-
-# Modify this function to download and re-upload the build image as needed
-def upload_build(source_build):
-    source_location = source_build['location']
-
-    # Modify as appropriate
-    dest_location = source_location
-    print("Uploading from %s to %s" % (source_location, dest_location))
-    # This placeholder function simply returns the unmodified source location.
-    # Re-uploading the build image is only necessary if the source and destination environments use different S3 buckets
-    return dest_location
-
-
-def create_promotion(_config, promotion):
-    dest = _config['dest']
-    
-    dest_auth = auth_context(dest['host'], dest['email'], dest['devkey'], dest['workspace'])
-
-    print("Starting promotion")
-    build_location = upload_build(promotion['build'])
-    if not dest['registered_model_name']:
-        model = create_model(dest_auth, promotion['model'], promotion['artifacts'])
-    else:
-        models = get_registered_models_by_name(dest_auth, dest['registered_model_name'])
-        if len(models) > 1:
-            print("WARNING: Multiple registered models with name '%s' found, using first one with id '%s'" % (dest['registered_model_name'], models[0]["id"]))
-        elif len(models) == 0:
-            print("ERROR: Registered model with name '%s' not found" % dest['registered_model_name'])
-            return
-        model = models[0]
-        print("Using existing registered model '%s'" % model['name'])
-    model_version = create_model_version(dest_auth, promotion['model_version'], model)
-
-    artifacts_and_model = promotion['artifacts']
-    artifacts_and_model.append(model_version['model'])
-    artifact_paths = upload_artifacts(dest_auth, model_version['id'], artifacts_and_model)
-
-    # Standard RMVs will have artifact path 'model' while ER->RMVs will have artifact path 'model.pkl'
-    model_artifact = model_version['model']
-    if 'model' in artifact_paths.keys():
-        model_artifact['path'] = artifact_paths['model']
-    else:
-        model_artifact['path'] = artifact_paths['model.pkl']
-    patch_model(dest_auth, model['id'], model_version['id'], model_artifact)
-
-    dest_build = create_build(dest_auth, model_version['id'], build_location, promotion['build'])
-    update_build_message(dest_auth, promotion['build'], dest_build)
-    return dest_build
-
-
-def promote(_config):
-    promotion = get_promotion_data(_config)
-    promoted_build = create_promotion(_config, promotion)
-    print("Promoted build %d created." % promoted_build['id'])
-
-
-promote(config)
+    print("Fetching or creating destination registered model %s" % source_model.name)
```

Comment:
If they provide `dest_model_id`, we should not ever create a destination model

Reply:
That's what the `else` is doing. I'm confused what this comment is suggesting to be done or what the issue is 🤔